pkg/bisect/bisect.go

```diff
@@ -85,11 +87,26 @@ func Run(cfg *Config) ([]*vcs.Commit, *report.Report, error) {
 	if !ok {
 		return nil, nil, fmt.Errorf("bisection is not implemented for %v", cfg.Manager.TargetOS)
 	}
+	commitEnv, ok := repo.(vcs.EnvForCommitOverridable)
+	if !ok {
+		return nil, nil, fmt.Errorf("repo does not implement EnvForCommit")
+	}
 	env := &env{
-		cfg:      cfg,
-		repo:     repo,
-		bisecter: bisecter,
+		cfg:       cfg,
+		repo:      repo,
+		bisecter:  bisecter,
+		commitEnv: commitEnv,
+	}
+	if env.inst, err = instance.NewEnv(&cfg.Manager); err != nil {
+		return nil, nil, err
+	}
+	if env.head, err = env.repo.CheckoutBranch(cfg.Kernel.Repo, cfg.Kernel.Branch); err != nil {
```

Comment:
We don't assign env.head in RunImpl nor in the test (?).
I think this function should do CheckoutBranch and ignore the commit and then RunImpl should obtain current Head from the repo and store in env.head.